lib/Onyx.js

```diff
@@ -1025,12 +1028,19 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * You may pass in keyStatesToPreserve to preserve Onyx keys you don't want to clear.
+ *
+ * @param {Array} keyStatesToPreserve ONYXKEYS values you want to keep
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(keyStatesToPreserve = []) {
     return getAllKeys()
         .then((keys) => {
             _.each(keys, (key) => {
+                if (_.contains(keyStatesToPreserve, key)) {
```

Comment:
This can work, but I thought the idea going forward was to prevent default key states and keys to preserve from being cleared from storage.

Reply:
There's just one issue though. For some reason the Storage assertion in the Native Storage test is _not_ passing while the Web version is 🤔 

Have to log off for a bit but will try to come back to this and see what's what.

Reply:
I think clearing the storage and then calling multiSet is a good approach.